src/jit/compile.c

```diff
@@ -123,7 +139,11 @@ MVMJitCode * MVM_jit_compile_graph(MVMThreadContext *tc, MVMJitGraph *jg) {
 }
 
 void MVM_jit_destroy_code(MVMThreadContext *tc, MVMJitCode *code) {
+#ifdef HAVE_LIBFFI
+    ffi_closure_free(code->writable_memory);
+#else
     MVM_platform_free_pages(code->func_ptr, code->size);
+#endif
```

Comment:
Correct, only the writable memory needs to be freed, not the exec memory, as they are actually the same memory just accessed via different mappings.

The exec memory and the writable memory are actually each just a base address and an offset into a mapping created via `mmap`.  These mappings are persistent across the life of the program.

`ffi_closure_alloc` uses a special allocator, `dlmalloc`, to manage memory via the writable mapping.  Memory is only allocated once but is visible through the exec mapping as well as through the writable mapping.  `ffi_closure_free` frees the memory via the writable mapping by calling `dlfree` at which point the exec memory address should no longer be used since it points (via the exec mapping) to memory which has been freed.